Fix howto in EGW docs

sridhartigera opened the pull request:
Fix EGW docs
1. Remove links from howto
2. copy files to CE and CC 3.16

ctauchen committed:
Fixes broken links for some ps1 files

ctauchen committed:
Updates to robots.txt

sridhartigera committed:
Fix EGW docs and copy to CC 3.16, CE 3.16

doucol committed:
add .netlify to gitignore [skip ci]

bmckercher123 reviewed (dismissed):
LGTM

bmckercher123 committed:
Add-warning-allow-tigera tier (#419)

Commit:
Fix dead links (#424)

doucol committed:
add tagline

bmckercher123 committed:
Fixes (#426)

bmckercher123 committed:
Update-graphic (#428)

doucol committed:
re-enable the subtitle / tagline for home page

hjiawei committed:
Remove alert forwarding not supported note (#432)

This changeset removes alerts forwarding not supported notes in
Calico Enterprise and Calico Cloud root, v3.15 and v3.16 docs.
Calico Enterprise do support alert forwarding.  It also reformats
`README.md` code blocks.

doucol committed:
add make index, add to sem job

doucol committed:
attach netlify secrets to sem

bmckercher123 reviewed (dismissed):
LGTM

bmckercher123 committed:
Update-image (#430)

ctauchen commented on calico-enterprise/networking/egress/egress-gateway-aws.mdx:
```diff
@@ -361,6 +361,11 @@ of resources). In addition, to support elastic IPs, each {{noderunning}} also re
 
+### Configure AWS Security Group rules
+
+In order to allow Egress Gateway traffic into the Egress Gateway pod's host from the client, the ingress rules of the security
```
```suggestion
In order to allow egress gateway traffic into the egress gateway pod's host from the client, the ingress rules of the security
```

ctauchen commented on calico-enterprise/networking/egress/egress-gateway-aws.mdx:
```diff
@@ -361,6 +361,11 @@ of resources). In addition, to support elastic IPs, each {{noderunning}} also re
+### Configure AWS Security Group rules
+
+In order to allow Egress Gateway traffic into the Egress Gateway pod's host from the client, the ingress rules of the security
+group needs to be updated. A rule to allow all packets from within the security group must be added to the inbound rules.
```
```suggestion
group need to be updated. A rule to allow all packets from within the security group must be added to the inbound rules.
```

ctauchen commented on calico-enterprise/networking/egress/egress-gateway-aws.mdx:
```diff
@@ -407,6 +412,7 @@ kubectl patch felixconfiguration default --type='merge' -p \
 
 - `egressIPSupport` must be the same on all cluster nodes, so you should set them only in the
   `default` FelixConfiguration resource.
+- As of v3.16.1, the operator will automatically enable the required policy sync API in the FelixConfiguration.
```
```suggestion
- The operator automatically enables the required policy sync API in the FelixConfiguration.
```
Best not to talk about specific version numbers, particularly when we're talking about future capabilities. If this is landing in 3.16.1, we can consider leaving this out for now and adding that information along with the release.

ctauchen commented on calico-enterprise/networking/egress/egress-gateway-on-prem.mdx:
```diff
@@ -202,10 +200,11 @@ kubectl patch felixconfiguration.p default --type='merge' -p \
 
 - `egressIPSupport` must be the same on all cluster nodes, so you should set them only in the
   `default` FelixConfiguration resource.
+- As of v3.16.1, the operator will automatically enable the required policy sync API in the FelixConfiguration. 
```
```suggestion
- As of v3.16.1, the operator will automatically enable the required policy sync API in the FelixConfiguration. 

:::
```
The second part of the note fence is missing.

ctauchen commented on calico-enterprise/networking/egress/external-network.mdx:
```diff
@@ -122,107 +122,49 @@ EOF
 ### Create egress gateways
 
-Create an egress gateway deployment running on the egress nodes and use new annotation `egress.projectcalico.org/externalNetworkNames` to associate the egress gateway pods of the deployment with one or more networks as desired. [see our other egress gateway guides](./)
+Use an Egress Gateway custom resource to deploy a group of egress gateways, and use `externalNetworks` to associate the egress gateway pods of the deployment with one or more networks as desired. [see our other egress gateway guides](./)
```
s/Egress Gateway/egress gateway

We do not capitalize egress gateway. Please make this change throughout.

ctauchen reviewed (commented):
Thanks for this -- a few comments in the code. For each, check for other instances of the same comment elsewhere in the PR.

This is mostly good to go. 

Housekeeping:
- please squash your commits
- needs approval from someone in CC to confirm these are all going there with the other 3.16 material CC @tmjd

doucol committed:
upgrade playwright, add rate limiting to redirect tests [skip ci]

doucol committed:
fail the redirect test if there are failures [skip ci]

tmjd commented on calico-cloud/networking/egress/egress-gateway-azure.mdx:
````diff
@@ -0,0 +1,678 @@
+To configure a node to be a route reflector with cluster ID 244.0.0.1, run the following command:
+
+```batch
+calicoctl patch node <node name> -p '{"spec": {"bgp": {"routeReflectorClusterID": "244.0.0.1"}}}'
````
This will not work for Calico Cloud, we do not make the calicoctl command available to cloud users.

sridhartigera replied:
Had a chat with @mazdakn. Looks like ```calicoctl``` is the only way to configure route reflector. Since calicoctl is not available to cloud users, I think EGW on azure won't work for calico cloud. I feel its better to remove the azure doc for calico cloud. WDYT?

tmjd commented on calico-cloud/networking/egress/egress-gateway-azure.mdx:
````diff
@@ -0,0 +1,678 @@
+Route Servers. To enable BGP routing protocol, set `bgp` field to `Enabled` in the default installation:
+
+```batch
+kubectl patch installation default --type='merge' -p '{"spec": {"calicoNetwork": {"bgp": "Enabled"}}}'
````
This won't work for AKS clusters that had AKS managed Calico and then were upgraded to Enterprise or Calico Cloud. For those installations the name of the installation resource would be `overlay`.
I don't know if in general the docs handle adjusting the name for the installation resource for clusters that use the `overlay` resource.

tmjd commented on calico-cloud/networking/egress/egress-gateway-azure.mdx:
````diff
@@ -0,0 +1,678 @@
+Or for support both per-namespace and per-pod:
+
+```batch
+kubectl patch felixconfiguration default --type='merge' -p \
````
We should be consistent with how we specify resources. Here we have `felixconfiguration`, a few lines above you have `felixconfiguration.p`.

bmckercher123 committed:
Fix alignment (#429)

hjiawei committed:
Add Tigera operator PSP step in docs (#434)

When Calico Enterprise v3.15+ is installed on a cluster with k8s <=
v1.24 and the admission controller is enabled, we need this additional
step to apply Tigera Operator PSP before applying the Tigera Operator
manifest.

bmckercher123 committed:
Add fips doc (#433)

doucol committed:
add rate limit to all tests [skip ci]

sridhartigera committed:
Merge branch 'main' into egw-howto

ctauchen commented on calico-cloud/networking/egress/external-network.mdx:
```diff
@@ -0,0 +1,182 @@
+---
```
@tmjd Just wanted to double-check: External networks is good to go for CC/3.16?

ctauchen commented on sidebars-calico-cloud.js:
```diff
@@ -345,8 +345,10 @@ module.exports = {
           link: {type: 'doc', id: 'networking/egress/index'},
           items: [
             'networking/egress/egress-gateway-on-prem',
+            'networking/egress/egress-gateway-azure',
```
I don't think this belongs, @sridhartigera

ctauchen commented on calico-enterprise_versioned_docs/version-3.16/networking/egress/external-network.mdx:
```diff
@@ -122,107 +122,49 @@ EOF
 
 Where:
-- Annotations `egress.projectcalico.org/externalNetworkNames: "[\"red\"]"` associates this egress gateway deployments with external network `red`. All traffic leaving the egress gateway to an external destination will be routed to the destination on external network `red`.
+- Where externalNetworks associates this egress gateway deployments with external network `red`. All traffic leaving the egress gateway to an external destination will be routed to the destination on external network `red`.
```
```suggestion
- Where `externalNetworks` associates this egress gateway deployments with external network `red`. All traffic leaving the egress gateway to an external destination will be routed to the destination on external network `red`.
```

sridhartigera committed:
Update calico-enterprise_versioned_docs/version-3.16/networking/egress/external-network.mdx

Co-authored-by: Christopher Tauchen <tauchen@gmail.com>

sridhartigera committed:
Remove EGW azure from CC

ctauchen commented:
Cloned in https://github.com/tigera/docs/pull/445
